yt_dlp/extractor/ondemandkorea.py

```diff
@@ -1,87 +1,164 @@
+                note='Checking if higher quality format is available', fatal=False) else url
+            for url in potential_urls
+        ]
+
+        formats = []
+        for url in potential_urls:
+            formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
+
+        subtitles = {}
+        for track in traverse_obj(data, ('result', 'text_tracks', lambda _, v: url_or_none(v['url']))):
+            subtitles.setdefault(track.get('language', 'und'), []).append({
+                'url': track['url'],
+                'ext': track.get('codec'),
+                'name': track.get('label'),
+            })
+
+        return {
+            'id': video_id,
+            'title': join_nonempty(
+                ('result', 'episode', 'program', 'title'),
+                ('result', 'episode', 'title'), from_dict=data, delim=': '),
+            **traverse_obj(data, ('result', {
+                'thumbnail': ('episode', 'images', 'thumbnail', {url_or_none}),
+                'release_date': ('episode', 'release_date', {lambda x: x.replace('-', '')}, {unified_strdate}),
+                'duration': ('duration', {functools.partial(float_or_none, scale=1000)}),
+                'age_limit': ('age_rating', 'name', {lambda x: x.replace('R', '')}, {parse_age_limit}),
+                'series': ('episode', {lambda x: x['program'] if x['kind'] == 'series' else None}, 'title'),
+                'series_id': ('episode', {lambda x: x['program'] if x['kind'] == 'series' else None}, 'id'),
+                'episode': ('episode', {lambda x: x['title'] if x['kind'] == 'series' else None},),
+                'episode_number': ('episode', {lambda x: x['number'] if x['kind'] == 'series' else None}, {int_or_none}),
+            }), get_all=False),
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+
+class OnDemandKoreaProgramIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ondemandkorea\.com/(?:en/)?player/vod/(?P<id>[a-z0-9-]+)(?:$|#)'
+    _GEO_COUNTRIES = ['US', 'CA']
+
+    _TESTS = [{
+        'url': 'https://www.ondemandkorea.com/player/vod/uskn-news',
+        'info_dict': {
+            'id': 'uskn-news',
+        },
+        'playlist_mincount': 755,
+    }, {
+        'url': 'https://www.ondemandkorea.com/en/player/vod/the-land',
+        'info_dict': {
+            'id': 'the-land',
+        },
+        'playlist_count': 52,
+    }]
+
+    _PAGE_SIZE = 100
+
+    def _fetch_page(self, display_id, page):
+        page += 1
+        page_data = self._download_json(
+            f'https://odkmedia.io/odx/api/v3/program/{display_id}/episodes/', display_id,
+            headers={'service-name': 'odk'}, query={
+                'page': page,
+                'page_size': self._PAGE_SIZE,
+            }, note=f'Downloading page {page}')
+        for episode in traverse_obj(page_data, ('result', 'results')):
```

Comment:
branch here to ensure list

you could filter an `'id'` key instead of `...`, if you want, depending if you want the extractor to error or not when it's not found
```suggestion
        for episode in traverse_obj(page_data, ('result', 'results', ...)):
```